src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,26 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  setSelectTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, setSelectTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(todo?.userId).then(setUser);
+  }, []);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
         <Loader />
       ) : (
```

Comment:
Create a loading state so you can show the loader depends on it, as in your case if something goes wrong and you don't have a user, the user see the infinite loader